api/src/main/java/io/jsonwebtoken/ClaimsMutator.java

```diff
@@ -124,10 +124,10 @@ public interface ClaimsMutator<T extends ClaimsMutator<T>> {
      * @param exp the JWT {@code exp} value or {@code null} to remove the property from the JSON map.
      * @return the {@code Claims} instance for method chaining.
      * @deprecated since 0.12.0 in favor of the shorter and more modern builder-style named
-     * {@link #expiration(Date)}. This method will be removed before the JJWT 1.0 release.
+     * {@link #expiration(Instant)}. This method will be removed before the JJWT 1.0 release.
      */
     @Deprecated
-    T setExpiration(Date exp);
+    T setExpiration(Instant exp);
```

Comment:
Keep the old signatures for the deprecated methods

Reply:
this data change requires is a major release, I would actually opt to remove these deprecated methods (in another pull request)

I modified these just to ensure it all compiles and not longer requires `java.time.Date`

Reply:
I get that, but the method is marked as deprecated, so that _shouldn't_ be changed.
(this is basically removing a old deprecated method, and adding a new deprecated method)

In general (and when possible) we want to provide an easy upgrade path for users, (e.g. adding an extra setter and deprecating the old one).
That's not possible with getters, (which would be a breaking change).

Reply:
the deprecated methods haven been reverted to use `java.util.Date`